### PR TITLE
Update Postgres schema

### LINKED ISSUE #43: Remove document state

### PR DESCRIPTION
**Pull request by calebschoepp**
# Overview

Update Postgres schema to reflect conversations we had in June 3 meeting.

# Changes

- Removed concept of `document_state`
- Changed `resume_review_states` table into a `resume_review_state` enum and `roles` table into `role_type` enum. I did this because:
    - Users don't need to add new types so no need make it INSERTABLE like a table
    - Enum is simpler than inner joins
    - Gives us nicer typing with Zapatos i.e. we will have strict typing in the service on the names of the states and roles
- Add `IF NOT EXISTS` in a number of places to make it more idempotent
- Update Zapatos schema

# Questions & Notes

- This is not standard practice to modify an existing migration file. I really should create a new one - something like `002_remove_document_state`. I'm not doing this b/c sticking to one migration file right now simplifies things and we have no production DB to worry about yet.
- ~I want #36 to merge before this~ merged

# Issue tracking

Closes #43 

# Testing & Demo

- Verified that up and down migrations successfully runned
- Inspected what Zapatos schema would look like with enum

### CODE REVIEW

File: api/src/zapatos/.eslintrc.json [hunk 1]
```diff
@@ -2,4 +2,4 @@
   "ignorePatterns": [
     "*"
   ]
-}
+}
```
Comment (calebschoepp) [inline api/src/zapatos/.eslintrc.json:5]: Want to keep this `No newline` here b/c it is auto-generated and we'll keep running into this issue.

File: api/db/migrations/001_initial_schema.down.sql [hunk 2]
```diff
@@ -10,17 +10,15 @@ DROP TRIGGER IF EXISTS update_time_documents ON documents;
...
 
 DROP TABLE IF EXISTS resume_reviews;
 
-DROP TABLE IF EXISTS resume_review_states;
+DROP TYPE IF EXISTS resume_review_state;
 
 DROP TABLE IF EXISTS user_roles;
 
-DROP TABLE IF EXISTS roles;
+DROP TYPE IF EXISTS role_type;
```
Comment (armiantos) [inline api/db/migrations/001_initial_schema.down.sql:21]: Any reason we're using `role_type` instead of just `role` since it's already a `TYPE`?

**Review by armiantos — COMMENTED** [review]
Overall LGTM, let me know what you think of my one comment lol

File: api/db/migrations/001_initial_schema.down.sql (hunk 2 above)
Comment (calebschoepp) [reply api/db/migrations/001_initial_schema.down.sql:21]: I didn't want it to conflict with field names like `user_roles.role`. Open to a better name

Comment (armiantos) [reply api/db/migrations/001_initial_schema.down.sql:21]: Good point, let's keep this 👍

**Review by armiantos — APPROVED** [review]
LGTM! Learning more about back-end everyday

**Pull request merged**